### PR TITLE
 Add EntityBaseFieldInfoAlterEvent

### PR DESCRIPTION
**Pull request by rmbom**
Added event to replace hook_entity_base_field_info_alter()

### CODE REVIEW

File: tests/src/Unit/EntityType/EntityBaseFieldInfoAlterEventTest.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    $this->manager = new HookEventDispatcherManagerSpy();
+    $builder->set('hook_event_dispatcher.manager', $this->manager);
+    $builder->compile();
+    \Drupal::setContainer($builder);
+  }
+
+  /**
+   * Test the EntityBaseFieldInfoEvent.
+   */
+  public function testEntityBaseFieldInfoEvent() {
```
Comment (pdenooijer) [inline tests/src/Unit/EntityType/EntityBaseFieldInfoAlterEventTest.php]: This test misses the real life usage of this event, please add an extra field to the fields after the event is fired. You can look up how I do this with the HookEventDispatcherManagerSpy in other tests. You could have a look at the `ViewEventTest` for instance.

**Review by pdenooijer — CHANGES_REQUESTED** [review]
First of all, thanks for your PR! To merge this, please fix the PHPCS errors and squash your commits to one commit 🙂.

**Comment by rmbom** [comment]
Done.

File: hook_event_dispatcher.module [hunk 2]
```diff
@@ -354,6 +355,19 @@ function hook_event_dispatcher_entity_base_field_info(EntityTypeInterface $entit
...
+ * Implements hook_entity_base_field_info_alter().
+ *
+ * {@inheritdoc}
+ */
+function hook_event_dispatcher_entity_base_field_info_alter(array &$fields, EntityTypeInterface $entityType) {
+  /* @var \Drupal\hook_event_dispatcher\Manager\HookEventDispatcherManagerInterface $manager */
+  $manager = \Drupal::service('hook_event_dispatcher.manager');
+  $event = new EntityBaseFieldInfoAlterEvent($fields, $entityType);
+  $manager->register($event);
+  return $event->getFields();
```
Comment (pdenooijer) [inline hook_event_dispatcher.module]: Ah sorry mist this in the first round. My last nitpick, please remove this line as the by reference fields is used and not the returned array.

**Review by pdenooijer — CHANGES_REQUESTED** [review]

**Comment by rmbom** [comment]
- Removed returning of the fields
- Expanded Unittest with check for getEntityType() for full coverage

**Review by pdenooijer — APPROVED** [review]

**Comment by pdenooijer** [comment]
Thnx 👍

**Comment by pdenooijer** [comment]
Dev version on drupal.org has your changes 👍

**Pull request merged**